CONVENTIONS.md

```diff
@@ -21,13 +21,37 @@ Additional conventions can be created by creating PRs which modify this document
 [Plugin specific fields](https://github.com/containernetworking/cni/blob/master/SPEC.md#network-configuration) formed part of the original CNI spec and have been present since the initial release.
 > Plugins may define additional fields that they accept and may generate an error if called with unknown fields. The exception to this is the args field may be used to pass arbitrary data which may be ignored by plugins.
 
-A plugin can define any additional fields it needs to work properly. It is expected that it will return an error if if can't act on fields that were expected or where the field values were malformed.
+A plugin can define any additional fields it needs to work properly. It is expected that it will return an error if it can't act on fields that were expected or where the field values were malformed.
 
-This method of passing information to a plugin is recommended when the information should be acted on and has specific meaning to that plugin.
+This method of passing information to a plugin is recommended when the information has specific meaning to the plugin that it's passed to and when the plugin should act on it or return an error if it can't.
```

Comment:
Yes, it's very convoluted! I think it wasn't clear before and maybe I've just made it worse now.

The goal of this paragraph is to give specific guidance on when this form of passing data to a plugin should be used. It aims to contrast this approach with the other two options. So the point I want to get across, is that this approach is used when the plugin is expected to act on the data. This is in contrast with the "args" approach where the data is much more loosely coupled and the runtime doesn't really know or care if the plugin acts on it.